auth-passwd.c

```diff
@@ -282,12 +282,33 @@ sys_auth_passwd(struct ssh *ssh, const char *password)
 		goto done;
 	}
 
-	if ((tmp = wcschr(user_utf16, L'@')) != NULL) {
-		udom_utf16 = tmp + 1;
+	/* logon user will accept principal name as user name with no domain 
+	 * but when backslash is present, we separate out the domain name. 
+	 * some identities such as internet-linked identities must be passed
+	 * in username@domain format to logon properly */
+	if ((tmp = wcschr(user_utf16, L'\\')) != NULL) {
+
+		/* break up the format domain\user into domain and user components*/
+		int uname_utf16_len = wcslen(user_utf16) + 1;
 		*tmp = L'\0';
+		udom_utf16 = user_utf16;
+		uname_utf16 = tmp + 1;
+
+		/* if the domain looks fully qualified, put in user@domain format */
+		if (wcschr(udom_utf16, L'.') != NULL) {
+			wchar_t upn[1024 + 1];
```

Comment:
Looks like this can be done irrespective of whether domain is fully-qualified/not. That way we can simplify and get rid of udom_utf16 altogether.

Reply:
I think this depends.  Do you consider passing username@netbios acceptable?  It may work empirically, but I don't think it's published as being a equivalent to UPN, is it?